src/app/map/components/GoogleMaps/index.tsx

```diff
@@ -1,87 +1,151 @@
 "use client";
 
-import React, { useState, useEffect, useMemo } from "react";
+import React, { useState, useEffect } from "react";
 import { useMantineColorScheme } from "@mantine/core";
 import { mapTheme, loader } from "~/utils";
 import Calendar from "../Calendar";
 import EventCard from "../EventCard";
-import { techEvents } from "~/events";
 import { useStyles } from "./styles";
-import useFetchEvent from "~/hooks/useFetchEvent";
+
+import { dummyData } from "../../event";
+import { Image, Text } from "@mantine/core";
+import {
+  GoogleMap,
+  LoadScript,
+  Marker,
+  InfoWindow,
+} from "@react-google-maps/api";
+
+export interface EventInterface {
+  id: string;
+  title: string;
+  eventUrl: string;
+  description: string;
+  venue?: {
+    id?: string;
+    name?: string;
+    address?: string;
+    lat?: number;
+    lng?: number;
+  } | null;
+  imageUrl: string;
+  imageId: string;
+  dateTime: string;
+}
+
+const containerStyle = {
+  width: "100vw",
+  height: "100vh",
+};
+
 
 interface GoogleMapsProps {
   setMapLoaded: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
 export const GoogleMaps = ({ setMapLoaded }: GoogleMapsProps) => {
-  const today: Date | null = useMemo(() => {
-    return new Date(Date.now());
-  }, []);
-  const [, setMap] = useState<google.maps.Map>();
-  const [date, setDate] = useState<Date | null>(today);
+
   const { colorScheme } = useMantineColorScheme();
   const { classes } = useStyles();
-  const { loading, error, result } = useFetchEvent(null, date);
-
-  console.log(loading, error, result);
+  const [infoWindowID, setInfoWindowID] = useState<number | null>(null);
+  const [currentMarker, setCurrentMarker] = useState<JSX.Element | null>(null);
+  const [markers, setMarkers] = useState<JSX.Element[] | null>(null); // Initialize as an empty array
+  const [date, setDate] = useState<Date | null>(new Date(Date.now()));
+  const Loadings = (
+    <div className={classes.gifWrapper}>
+      <Image
+        width={150}
+        height={150}
+        mt={-80}
+        src="/images/loading.gif"
+        alt="loading"
+      />
+      <Text align="center" fz="sm" mt={15}>
+        searching for <br />
+        upcoming events <br />
+        near you ...
+      </Text>
+    </div>
+  );
```

Comment:
Can we use Joy's loading component instead of creating new one?

Reply:
I think I used Joy's loading component but I had to integrated it in the GoogleMaps component....

Reply:
huum, my suggestion was importing Loading component rather than copying it like:
```typescript
import Loading from "./Loading" 
// some codes
<LoadScript
        googleMapsApiKey={process.env.NEXT_PUBLIC_GOOGLE_MAPS_KEY || ''}
        loadingElement={<Loading />} // here
      >
```
Was it impossible?